playbooks/os/openshift-cluster/config.yml

```diff
@@ -0,0 +1,39 @@
+---
+# TODO: need to figure out a plan for setting hostname, currently the default
+# is localhost, so no hostname value (or public_hostname) value is getting
+# assigned
```

Comment:
@detiber:
Here is the issue that I have now with my OpenStack setup:

When the VMs are created, their hostname is initially the symbolic name (eg.: lenaic-master-1c577.novalocal) (passed by OpenStack to the VM via the metadata)

##### If I put `openshift_hostname: "{{ ansible_default_ipv4.address }}"` in `playbooks/openstack/openshift-cluster/config.yml` (similar to what is done for GCE)

The `Set hostname` task of `roles/openshift_common/tasks/main.yml` changes the hostname to use the IP (eg.: 192.168.211.2).
The result is that all machines have, as hostname, 192, and all machines have a different domain, (eg.: 168.211.2)
If the machines are rebooted, they get their symbolic name (lenaic-master-1c577.novalocal) as hostname again.

##### If I put `openshift_public_hostname: "{{ ansible_default_ipv4.address }}"` in `playbooks/openstack/openshift-cluster/config.yml` (similar to what is done for AWS)

The machines are correctly named with their symbolic name. But the machines are referenced by name from other and because there has been no DNS registration, the name resolution fails

```
[openshift@lenaic-master-f1d2d ~]$ osc get nodes
NAME                          LABELS        STATUS
lenaic-node-64c82.novalocal   Schedulable   <none>    NotReady
lenaic-node-a5fc9.novalocal   Schedulable   <none>    NotReady
```
* hostname of the master is correct (in the prompt)
* name of the nodes are correct
* The nodes are not ready because DNS resolution doesn’t work.

Ex of logs:
```
Jun 02 04:10:54 lenaic-master-f1d2d.novalocal openshift-sdn-master[12945]: E0602 04:10:54.498659 12945 controller.go:129] Failed to lookup IP address for minion lenaic-node-64c82.novalocal.
```

##### If I set neither `openshift_hostname` nor `openshift_public_hostname` in `playbooks/openstack/openshift-cluster/config.yml` (similar to what is done for libvirt)

Same issue as above: the hostname of the machines is correctly the symbolic name bus as the DNS resolution doesn’t work, symbolic names cannot be used on the master

```
[openshift@lenaic-master-5947c ~]$ osc get nodes
NAME                          LABELS        STATUS
lenaic-node-13248.novalocal   Schedulable   <none>    NotReady
lenaic-node-580e1.novalocal   Schedulable   <none>    NotReady
```

Ex of logs:
```
Jun 02 04:51:28 lenaic-master-5947c.novalocal openshift-sdn-master[12928]: E0602 04:51:28.281472 12928 controller.go:129] Failed to lookup IP address for minion lenaic-node-13248.novalocal.
```

##### Conclusion

I wish I could put neither `openshift_hostname` nor `openshift_public_hostname` in `config.yml` because the machines already know their names thanks to metadata.
The issue I have with that is that machines reference each other via DNS names and DNS resolution doesn’t work.
Ex:
```
ps:
/usr/bin/openshift-sdn -minion -etcd-endpoints=https://lenaic-master-5947c.novalocal:4001
logs:
… lenaic-node-13248.novalocal openshift-sdn-node[12972]: … 501: All the given peers are not reachable 
```

Putting neither `openshift_hostname` nor `openshift_public_hostname` works for the libvirt setup because libvirt uses a dnsmasq to manage the VMs’ network and that dnsmasq is used as both DHCP and DNS server and dnsmasq’s DHCP leases are published in dnsmasq’s DNS.

So, for OpenStack, I would like to put IP everywhere, except for the hostnames.